// @openmeteo/sdk/weather_api.js
"use strict";
// automatically generated by the FlatBuffers compiler, do not modify
Object.defineProperty(exports, "__esModule", { value: true });
exports.WeatherApiResponse = exports.VariablesWithTime = exports.VariableWithValues = exports.Variable = exports.Unit = exports.Model = exports.Aggregation = void 0;
var aggregation_js_1 = require("./aggregation.js");
Object.defineProperty(exports, "Aggregation", { enumerable: true, get: function () { return aggregation_js_1.Aggregation; } });
var model_js_1 = require("./model.js");
Object.defineProperty(exports, "Model", { enumerable: true, get: function () { return model_js_1.Model; } });
var unit_js_1 = require("./unit.js");
Object.defineProperty(exports, "Unit", { enumerable: true, get: function () { return unit_js_1.Unit; } });
var variable_js_1 = require("./variable.js");
Object.defineProperty(exports, "Variable", { enumerable: true, get: function () { return variable_js_1.Variable; } });
var variable_with_values_js_1 = require("./variable-with-values.js");
Object.defineProperty(exports, "VariableWithValues", { enumerable: true, get: function () { return variable_with_values_js_1.VariableWithValues; } });
var variables_with_time_js_1 = require("./variables-with-time.js");
Object.defineProperty(exports, "VariablesWithTime", { enumerable: true, get: function () { return variables_with_time_js_1.VariablesWithTime; } });
var weather_api_response_js_1 = require("./weather-api-response.js");
Object.defineProperty(exports, "WeatherApiResponse", { enumerable: true, get: function () { return weather_api_response_js_1.WeatherApiResponse; } });
